aten/src/ATen/core/op_registration/kernel_function_test.cpp

```diff
@@ -537,4 +540,308 @@ TEST(OperatorRegistrationTest_FunctionBasedKernel, givenKernelWithTensorListInpu
   EXPECT_EQ(2, outputs[0].toInt());
 }
 
+template<class Return, class... Args> struct kernel_func final {
+  static Return func(Args...) { return {}; }
+};
+template<class... Args> struct kernel_func<void, Args...> final {
+  static void func(Args...) {}
+};
+
+TEST(OperatorRegistrationTest_FunctionBasedKernel, givenMismatchedKernel_withDifferentNumArguments_whenRegistering_thenFails) {
+  // assert this does not fail because it matches
+  RegisterOperators()
+      .op(FunctionSchema(
+          "_test::mismatch",
+          "",
+          (std::vector<Argument>{Argument("arg")}),
```

Comment:
I'd prefer that we do not generate a bunch of test cases directly using the Argument and FunctionSchema constructors for the same reason it is not a good idea for client code to us them.

Reply:
that's the only API that works as long as I don't have the function schema parser in c10. I'll change all these test cases towards taking a string once the API supports it. But for this diff, I rather have some test cases with a provisional API than no test cases at all.